base/inference.jl

```diff
@@ -1397,10 +1395,6 @@ function abstract_call_gf_by_type(f::ANY, atype::ANY, sv::InferenceState)
         add_mt_backedge(ftname.mt, argtype, sv)
         update_valid_age!(min_valid[1], max_valid[1], sv)
     end
-    if isempty(applicable)
-        # TODO: this is needed because type intersection is wrong in some cases
-        return Any
```

Comment:
Are we sure we are ready for this? Admittedly, of the `typeintersect` bugs I have seen recently, none was erroneously giving `Bottom`, but since proposing this in #20866, I have lost some confidence in `typeintersect`... Anyway, we can try, and if there are problems and we can't straighten it out in `subype.c` in time, we can just add this back.